core/block/export/export.go

```diff
@@ -608,12 +608,20 @@ func validType(sbType smartblock.SmartBlockType) bool {
 		sbType == smartblock.SmartBlockTypeFileObject
 }
 
-func validTypeForNonProtobuf(sbType smartblock.SmartBlockType) bool {
+func validTypeForNonProtobuf(sbType smartblock.SmartBlockType, details *types.Struct) bool {
+	if !validLayout(details) {
+		return false
+	}
 	return sbType == smartblock.SmartBlockTypeProfilePage ||
 		sbType == smartblock.SmartBlockTypePage ||
 		sbType == smartblock.SmartBlockTypeFileObject
 }
 
+func validLayout(details *types.Struct) bool {
```

Comment:
Maybe we should rename this func as validLayoutForNonProtobuf and move its invokation to line 541:

`if !isProtobuf && !validTypeForNonProtobuf(sbType) && !validLayoutForNonProtobuf(r.Details) {`

Reply:
Fixed